src/render/draw_sky.js

```diff
@@ -8,101 +8,52 @@ import Context from '../gl/context.js';
-    const gl = context.gl;
-    const transform = painter.transform;
-    const depthMode = new DepthMode(gl.LEQUAL, DepthMode.ReadOnly, [0, 1]);
-    const program = painter.useProgram('globeAtmosphere');
-
-    // Compute center and approximate radius of the globe on screen coordinates
-    const viewMatrix = transform._camera.getWorldToCamera(transform.worldSize, 1.0);
-    const viewToProj = transform._camera.getCameraToClipPerspective(transform._fov, transform.width / transform.height, transform._nearZ, transform._farZ);
-    const globeToView = mat4.mul([], viewMatrix, calculateGlobeMatrix(transform, transform.worldSize));
-    const viewToScreen = mat4.mul([], transform.labelPlaneMatrix, viewToProj);
-
-    const centerOnViewSpace = vec3.transformMat4([], [0, 0, 0], globeToView);
-    const radiusOnViewSpace = vec3.add([], centerOnViewSpace, [transform.worldSize / Math.PI / 2.0, 0, 0]);
-
-    const centerOnScreen = vec3.transformMat4([], centerOnViewSpace, viewToScreen);
-    const radiusOnScreen = vec3.transformMat4([], radiusOnViewSpace, viewToScreen);
-
-    const pixelRadius = vec3.length(vec3.sub([], radiusOnScreen, centerOnScreen));
-
-    const uniforms = atmosphereUniformValues(
-        centerOnScreen,
-        pixelRadius,
-        [transform.width, transform.height],
-        browser.devicePixelRatio,
-        1.0,                        // opacity
-        2.0,                        // fadeout range
-        [1.0, 1.0, 1.0],            // start color
-        [0.0118, 0.7451, 0.9882]);  // end color
-
-    painter.prepareDrawProgram(context, program);
-
-    const sharedBuffers = painter.globeSharedBuffers;
-    if (sharedBuffers) {
-        program.draw(context, gl.TRIANGLES, depthMode, StencilMode.disabled,
-            ColorMode.alphaBlendedReal, CullFaceMode.backCW, uniforms, "skybox",
-            sharedBuffers.atmosphereVertexBuffer,
-            sharedBuffers.atmosphereIndexBuffer,
-            sharedBuffers.atmosphereSegments);
-    }
-}
-
 const TRANSITION_OPACITY_ZOOM_START = 7;
 const TRANSITION_OPACITY_ZOOM_END = 8;
 
 function drawSky(painter: Painter, sourceCache: SourceCache, layer: SkyLayer) {
-    if (painter.transform.projection.name === 'globe') {
-        drawGlobeAtmosphere(painter);
-    } else {
-        const tr = painter.transform;
-        const isMercator = tr.projection.name === 'mercator';
-        // For non-mercator projection, use a forced opacity transition. This transition is set to be
-        // 1.0 after the sheer adjustment upper bound which ensures to be in the mercator projection.
-        const transitionOpacity = isMercator ? 1.0 : smoothstep(TRANSITION_OPACITY_ZOOM_START, TRANSITION_OPACITY_ZOOM_END, tr.zoom);
-        const opacity = layer.paint.get('sky-opacity') * transitionOpacity;
-        if (opacity === 0) {
-            return;
-        }
+    const tr = painter.transform;
+    const shouldTransition = (tr.projection.name === 'mercator' || tr.projection.name === 'globe');
```

Comment:
nit: I think the renaming messed the intent here, it's basically `shouldNotTransition`. We could just have `globeOrMercator`.

```suggestion
    const globeOrMercator = (tr.projection.name === 'mercator' || tr.projection.name === 'globe');
```

This is used only by non-mercator projections to make sure that we don't render sky layers at low zooms since they aren't supported by these projections. It's different from the actual globe / mercator transition.